src/Firebase/AppCheck/DecodedAppCheckToken.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use InvalidArgumentException;
+use JsonSerializable;
+
+use function array_key_exists;
+
+/**
+ * @phpstan-type DecodedAppCheckTokenShape array{
+ *     app_id: string,
+ *     aud: array<string>,
+ *     exp: string,
+ *     iat: string,
+ *     iss: string,
+ *     sub: string,
+ * }
+ */
+final class DecodedAppCheckToken implements JsonSerializable
+{
+    private string $app_id;
+
+    /** @var array<string> */
+    private array $aud;
+    private string $exp;
+    private string $iat;
+    private string $iss;
+    private string $sub;
+
+    /**
+     * @param array<string> $aud
+     */
+    private function __construct(
+        string $app_id,
+        array $aud,
+        string $exp,
+        string $iat,
+        string $iss,
+        string $sub,
+    ) {
+        $this->app_id = $app_id;
+        $this->aud = $aud;
+        $this->exp = $exp;
+        $this->iat = $iat;
+        $this->iss = $iss;
+        $this->sub = $sub;
+    }
+
+    /**
+     * @param array<string, mixed> $data
```

Comment:
```suggestion
     * @param DecodedAppCheckTokenShape $data
```

This would remove the need for the `array_key_exists` checks - it's a public function, but in reality, this will only be used by the SDK itself.